src/pages/CreateProposal/index.tsx

```diff
@@ -0,0 +1,225 @@
+
+  const [modalOpen, setModalOpen] = useState(false)
+  const [hash, setHash] = useState<string | undefined>()
+  const [attempting, setAttempting] = useState(false)
+  const [proposalAction, setProposalAction] = useState(ProposalAction.TRANSFER_TOKEN)
+  const [toAddressValue, setToAddressValue] = useState('')
+  const [currencyValue, setCurrencyValue] = useState<Currency>(UNI[chainId ?? 1])
+  const [amountValue, setAmountValue] = useState('')
+  const [titleValue, setTitleValue] = useState('')
+  const [bodyValue, setBodyValue] = useState('')
+
+  const handleActionSelectorClick = useCallback(() => [setModalOpen(true)], [setModalOpen])
+  const handleActionChange = useCallback(
+    (proposalAction: ProposalAction) => [setProposalAction(proposalAction)],
+    [setProposalAction]
+  )
+  const handleDismissActionSelector = useCallback(() => [setModalOpen(false)], [setModalOpen])
+  const handleDismissSubmissionModal = useCallback(
+    () => [setHash(undefined), setAttempting(false)],
+    [setHash, setAttempting]
+  )
+  const handleToAddressInput = useCallback((toAddress: string) => [setToAddressValue(toAddress)], [setToAddressValue])
+  const handleCurrencySelect = useCallback((currency: Currency) => [setCurrencyValue(currency)], [setCurrencyValue])
+  const handleAmountInput = useCallback((amount: string) => [setAmountValue(amount)], [setAmountValue])
+  const handleTitleInput = useCallback((title: string) => [setTitleValue(title)], [setTitleValue])
+  const handleBodyInput = useCallback((body: string) => [setBodyValue(body)], [setBodyValue])
+
+  const isFormInvalid = useMemo(
+    () =>
+      Boolean(
+        !proposalAction ||
+          !utils.isAddress(toAddressValue) ||
+          !currencyValue?.isToken ||
+          amountValue === '' ||
+          titleValue === '' ||
+          bodyValue === ''
+      ),
+    [proposalAction, toAddressValue, currencyValue, amountValue, titleValue, bodyValue]
+  )
+
+  const createProposalCallback = useCreateProposalCallback()
+
+  const handleCreateProposal = async () => {
+    setAttempting(true)
+
+    if (!createProposalCallback) return
+
+    const createProposalData: CreateProposalData = {} as CreateProposalData
+    let calldataValues: string[] = []
+
+    createProposalData.targets = currencyValue?.isToken ? [currencyValue.address] : []
+    createProposalData.values = ['0']
+
+    switch (proposalAction) {
+      case ProposalAction.TRANSFER_TOKEN:
+        createProposalData.signatures = ['transfer(address,uint256)']
+        calldataValues = [toAddressValue, amountValue]
+        break
+      case ProposalAction.APPROVE_TOKEN:
+        createProposalData.signatures = ['approve(address,uint256)']
+        calldataValues = [toAddressValue, amountValue]
+        break
+      default:
+        createProposalData.signatures = []
+    }
+
+    const typesArray = createProposalData.signatures[0].split('(').pop()?.split(')')[0]?.split(',')
```

Comment:
a bit too spooky, we should define these per-proposal action in the switch, also this is hard-coded to only work with a length-1 array, should at least have a nod toward it being generic.

the way to do this is probably something like:

```
types = [['address', 'uint256']]
createProposalData.signatures = [`transfer(${types[0].join(',')})`]
```